src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,48 @@
+package duke;
+
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+
+public class Duke {
+    private final Ui ui;
+    private final Storage storage;
+    private final TaskList task;
+    private final Parser parse;
+
+    public Duke(String filePath) {
+        ArrayList<Task> tasks = new ArrayList<>();
+        ui = new Ui();
+        storage = new Storage(filePath);
+        task = new TaskList(filePath, tasks);
+        parse = new Parser(filePath, tasks);
+        try {
+            storage.loadFile(task.getTask());
+        } catch (FileNotFoundException err) {
+            ui.showShortLine();
+            ui.fileNotFoundMessage();
+        }
+    }
+
+    public void run() {
+        ui.welcomeMessage();
+        if (task.getTaskSize() != 0) {
+            ui.taskLoadMessage();
+            System.out.println("There are " + task.getTaskSize() + " tasks present");
+        } else {
+            ui.fileEmptyMessage();
+        }
+        ui.showShortLine();
+        parse.processUserInput();
```

Comment:
gives me an impression that the parser may be doing too many things?
Parse should parse.
Processing the input is some one else's responsibility (you will learn about a design principle that talks abt this a little later in the semester)